aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -210,6 +210,12 @@ module aptos_framework::object {
         create_object_internal(creator_address, obj_addr, false)
     }
 
+    /// Create a new object by generating a random UUID based on transaction hash
+    public fun create_object(creator: &signer): ConstructorRef {
+        let uuid = aptos_framework::transaction_context::create_uuid();
+        create_object_internal(signer::address_of(creator), uuid, true)
```

Comment:
it seems like true is used whenever unique (non-repeatable) object address is used (i.e. in create_object_from_guid), and false otherwise.

if we want to allow users to specify if object can be deleted or not, that can be made a function argument, but that doesn't seem to be the current logic?

Reply:
@vusirikala  - let's make all functions except create_named_object and create_user_derived_object to use it (i.e. remove all uses of create_object_from_guid here, and remove that function.

Reply:
My point is to add documentation :) This type of object being deletable makes sense to me if we can guarantee the same address can never be regenerated with future txs.

Reply:
I made changes to create_object_from_object, create_object_from_account, and removed create_object_from_guid. I added documentation that the objects created from create_object can be deleted because the created uuids can never be regenerated with future txs.